docs: add documentation for demo code (#185)
stackabletech/demos: docs/modules/demos/pages/index.adoc

[IMPORTANT]
.External Components in these demos
====
The demos provided by Stackable serve as showcases to illustrate potential architectures that can be built using the Stackable Data Platform.
Please note that these demos may include components that are not supported by Stackable as part of our commercial offering.

If you are evaluating these demos with the intention of purchasing a subscription, please ensure that you review the xref:operators:index.adoc[list of supported operators].
Any components not listed there are not part of our commercial offering.

Below is an incomplete list of third-party components referenced in these demos:

* Grafana
* JupyterHub
* Keycloak
* MinIO
* OpenLDAP
* OpenSearch
* OpenSearch Dashboards
* PostgreSQL
* Prometheus
* Redis
* Vector Aggregator
====

== Demo Code

> The manifests and code used by the demos, can be found in the https://github.com/stackabletech/demos[demo repository,window=_blank]. Each demo is constructed in the following way:

Reviewer: nit
```suggestion
The manifests and code used by the demos, can be found in the https://github.com/stackabletech/demos[demo repository,window=_blank].
Each demo is constructed in the following way:
```

> * `demos/demos-v2.yaml` file: lists the available demos and uses additional parameters to build up the demo, e.g.

Reviewer: We also have https://docs.stackable.tech/management/stable/stackablectl/customization/, we should probably link to it

Author: Ah good to know about that section! https://github.com/stackabletech/demos/pull/185/commits/ac6cc7eca2b43fde74e025a698374c7713121d12

Reviewer: Thanks!

  ** `stackableStack` references the stack belonging to this demo
  ** `manifests` includes a list of additional manifests for the demo, mostly jobs to create tables or ingest example data
  ** `resourceRequests` specifying the needed resources for the demo to run

> * `stacks/stacks-v2.yaml` file: lists the available stacks, which are referenced by the `demos/demos-v2.yaml` file, and uses additional parameters to build up the demo, e.g.

Reviewer 2: ```suggestion
* `stacks/stacks-v2.yaml` file: lists the available stacks, which are referenced by the `demos/demos-v2.yaml` file, and uses additional parameters to build up the underlying stack, e.g.
```

* JupyterHub
* Keycloak
* MinIO
* OpenLDAP
* OpenSearch
* OpenSearch Dashboards
* PostgreSQL
* Prometheus
* Redis
* Vector Aggregator
====

== Demo Code

The manifests and code used by the demos, can be found in the https://github.com/stackabletech/demos[demo repository,window=_blank].
Each demo is constructed in the following way:

* `demos/demos-v2.yaml` file: lists the available demos and uses additional parameters to build up the demo, e.g.
  ** `stackableStack` references the stack belonging to this demo
  ** `manifests` includes a list of additional manifests for the demo, mostly jobs to create tables or ingest example data
  ** `resourceRequests` specifying the needed resources for the demo to run
* `stacks/stacks-v2.yaml` file: lists the available stacks, which are referenced by the `demos/demos-v2.yaml` file, and uses additional parameters to build up the underlying stack, e.g.
  ** `stackableOperators` are all the operators that are installed as part of the demo
  ** `manifests` contains all the manifests needed to setup the products and third-party tools
  ** `parameters` is a list of values used in the manifest files, for example as credentials or other settings

> For more information on how to add custom demos/stacks, refer to the xref:management:stackablectl:customization/index.adoc[Customization section of SDP Management,window=_blank].

Reviewer: ```suggestion
For more information on how to add custom demos/stacks, refer to the xref:management:stackablectl:customization/index.adoc[Customization section of stackablectl,window=_blank].
```
